### DISCUSSION TITLE
Embed static files and folders into binary and serve them by pocketbase

### DISCUSSION THREAD

**Discussion by vipas84** [pocketbase/pocketbase#4810] [Q&A] [answered]
Hello,
I have the following need:
- I have an SPA app (Svelte SPA) as a frontend for Pocketbase backend (as a franework).
- I need to embed in to a custom pocketbase binary (used as a framework) and serve them via pocketbase builtin static webserver

What could be the best approach to do so?

Thank in advance for your kind support

**Comment by ganigeorgiev** [comment, marked as answer]
You can use the Go [embed directive](https://pkg.go.dev/embed).

As a reference you can check the [PocketBase Admin UI embed.go](https://github.com/pocketbase/pocketbase/blob/master/ui/embed.go).

Once the embed variable are defined you can register a new route with the [`StaticDirectoryHandler`](https://pocketbase.io/docs/go-routing/#serving-static-files):

```js
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
    // DistDirFS is from embed.go
    e.Router.GET("/*", apis.StaticDirectoryHandler(DistDirFS, false))

    return nil
})
```

**Reply by vipas84** [reply]
Amazing! It works!
